Extensions/Share/ShareViewController.swift

```diff
@@ -0,0 +1,303 @@
+            label.lineBreakMode = .byTruncatingMiddle
+            label.font = UX.baseFont
+        }
+
+        pageTitleLabel.font = UIFont.boldSystemFont(ofSize: UX.baseFont.pointSize)
+
+        return (row, pageTitleLabel, urlLabel)
+    }
+
+    private func makeActionRow(label: String, imageName: String, action: Selector, hasNavigation: Bool) -> UIView {
+        let row = UIView()
+
+        let stackView = UIStackView()
+        stackView.axis = .horizontal
+        stackView.spacing = UX.actionRowSpacingBetweenIconAndTitle
+        row.addSubview(stackView)
+
+        stackView.snp.makeConstraints { make in
+            make.top.bottom.equalToSuperview()
+            make.left.right.equalToSuperview().inset(UX.pageInfoRowLeftInset)
+        }
+
+        let icon = UIImageView(image: UIImage(named: imageName))
+        icon.contentMode = .scaleAspectFit
+
+        let title = UILabel()
+        title.font = UX.baseFont
+        title.numberOfLines = 2
+        title.adjustsFontSizeToFitWidth = true
+        title.allowsDefaultTighteningForTruncation = true
+
+        title.text = label
+        [icon, title].forEach { stackView.addArrangedSubview($0) }
+
+        icon.snp.makeConstraints { make in
+            make.size.equalTo(UX.actionRowIconSize)
+        }
+
+        if hasNavigation {
+            let navButton = UIImageView(image: UIImage(named: "menu-Disclosure"))
+            navButton.contentMode = .scaleAspectFit
+            stackView.addArrangedSubview(navButton)
+            navButton.snp.makeConstraints { make in
+                make.size.equalTo(14)
+            }
+        }
+
+        let gesture = UITapGestureRecognizer(target: self, action: action)
+        row.addGestureRecognizer(gesture)
+
+        actionRows.append(row)
+        return row
+    }
+
+    fileprivate func animateToActionDoneView(withTitle title: String = "") {
+        navigationItem.leftBarButtonItem = nil
+
+        navigationController?.view.snp.updateConstraints {
+            make in
+            make.height.equalTo(UX.viewHeightForDoneState)
+        }
+
+        UIView.animate(withDuration: 0.2, animations: {
+            self.actionRows.forEach { $0.removeFromSuperview() }
+            self.separators.forEach { $0.removeFromSuperview() }
+            self.navigationController?.view.superview?.layoutIfNeeded()
+        }, completion: { _ in
+            self.showActionDoneView(withTitle: title)
+        })
+    }
+
+    @objc func finish(afterDelay: TimeInterval = UX.durationToShowDoneDialog) {
+        delegate?.finish(afterDelay: afterDelay)
+    }
+
+    private func showActionDoneView(withTitle title: String) {
+        let blue = UIView()
+        blue.backgroundColor = UX.doneLabelBackgroundColor
+        self.stackView.addArrangedSubview(blue)
+        blue.snp.makeConstraints { make in
+            make.height.equalTo(UX.pageInfoRowHeight)
+        }
+
+        let label = UILabel()
+        label.text = title
```

Comment:
I think both the label and the checkmark need a bold font. The font also looks a bit heavier in the mock

Reply:
I wonder if `UIFont.systemFont(ofSize: 18, weight: UIFont.Weight.heavy)` looks heavier than bold?